@xen-orchestra/proxy/src/app/mixins/reverseProxy.mjs

```diff
@@ -0,0 +1,124 @@
+    // keep redirect url relative to local server
+    const localPath = `${basePath}/${backendUrl.pathname.substring(target.pathname.length)}${backendUrl.search}${
+      backendUrl.hash
+    }`
+    return localPath
+  }
+
+  _getConfigFromRequest(req) {
+    for (const [path, config] of Object.entries(this._app.config.get('reverseProxies'))) {
+      const fullPath = '/proxy/v1/' + removeSlash(path)
+      if (req.url.startsWith(fullPath + '/')) {
+        if (typeof config === 'string') {
+          return {
+            path: fullPath,
+            target: config,
+            options: {},
+          }
+        }
+
+        const { target, ...options } = config
+        return {
+          path: fullPath,
+          target,
+          options,
+        }
+      }
+    }
+  }
+
+  proxy(req, res) {
+    const config = this._getConfigFromRequest(req)
+
+    if (config === undefined) {
+      return
+    }
+
+    const url = new URL(config.target)
+    const targetUrl = this._localToBackendUrl(config.path, url, req.originalUrl || req.url)
+    proxy.web(req, res, {
+      ...urlToHttpOptions(targetUrl),
+      ...config.options,
+      onReq: (req, { headers }) => {
+        headers['x-forwarded-for'] = req.socket.remoteAddress
+        headers['x-forwarded-proto'] = req.socket.encrypted ? 'https' : 'http'
+        if(req.headers['host'] !== undefined){
+          headers['x-forwarded-host'] = req.headers['host']
+        }
+      },
+      onRes: (req, res, proxyRes) => {
+        // rewrite redirect to pass through this proxy
+        if (proxyRes.statusCode === 301 || proxyRes.statusCode === 302) {
+          // handle relative/ absolute location
+          const redirectTargetLocation = new URL(proxyRes.headers.location, url)
+
+          // this proxy should only allow communication between known hosts. Don't open it too much
+          if (redirectTargetLocation.hostname !== url.hostname || redirectTargetLocation.protocol !== url.protocol) {
+            throw new Error(`Can't redirect from ${url.hostname} to ${redirectTargetLocation.hostname} `)
+          }
+          res.writeHead(proxyRes.statusCode, {
+            ...proxyRes.headers,
+            location: this._backendToLocalPath(config.path, url, redirectTargetLocation),
+          })
+          res.end()
+          return
+        }
+        // pass through the answer of the remote server
+        res.writeHead(proxyRes.statusCode, {
+          ...proxyRes.headers,
+        })
+        // pass through content
+        proxyRes.pipe(res)
+      },
+    })
+  }
+
+  upgrade(req, socket, head) {
+    const config = this._getConfigFromRequest(req)
+    if (config === undefined) {
+      return
+    }
+
+    const { path, target, options } = config
+    const targetUrl = this._localToBackendUrl(path, target, req.originalUrl || req.url)
+    proxy.ws(req, socket, head, {
+      ...urlToHttpOptions(targetUrl),
+      ...options,
+    })
```

Comment:
Is the redirection not relevant in that case?

Reply:
I don't know , you tell me

Reply:
I'm asking because I see you've decided to handle redirections in the other handler but not the one, why this choice?

Reply:
Probably useless in this case.